src/main/java/vct/col/newrewrite/SpecifyImplicitLabels.scala

```diff
@@ -0,0 +1,45 @@
+package vct.col.newrewrite
+
+import vct.col.ast.RewriteHelpers._
+import vct.col.ast._
+import vct.col.ast.AstBuildHelpers._
+
+import scala.collection.mutable
+
+case class SpecifyImplicitLabels() extends Rewriter {
+  val labelStack = new mutable.Stack[LabelDecl]()
+
+  override def dispatch(stat: Statement): Statement = stat match {
+    case block@Block(Seq(Label(labelDecl), _: Loop)) =>
+      labelStack.push(labelDecl)
+      val res = rewriteDefault(block)
+      labelStack.pop()
+      res
+    case block@Block(Seq(Label(labelDecl), _: Switch)) =>
+      labelStack.push(labelDecl)
+      val res = rewriteDefault(block)
+      labelStack.pop()
+      res
```

Comment:
This pattern occurs so much that I usually write a helper for it, see e.g. ScopeContext.collectInScope. Perhaps we should make a generic util version of this somewhere.

Reply:
It is now `hre.util.ScopedStack`, which you can use like:

```scala
val labelStack = ScopedStack[LabelDecl]()

labelStack.having(labelDecl) {
  rewriteDefault(block)
}
```